invenio_drafts_resources/services/records/service.py

```diff
@@ -181,19 +182,40 @@ def publish(self, id_, identity, links_config=None):
         """Publish a draft."""
         self.require_permission(identity, "publish")
 
-        # Get draft
+        # Get data layer draft
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
-
+        # At this point, draft has
+        # {..., "access": {'record': 'public', 'files': 'public', 'owned_by': [{'user': 1}]}}
+        print("REcordDraftService - publish - draft['access']", draft["access"])
+
+        # Validate draft for publication. This validation is stricter than
+        # the one for saving.
+        # 1- Get projected draft i.e. service layer draft result item
+        draft_item = self.result_item(
+            self, identity, draft, links_config=links_config
+        )
+        print("RecordDraftService - publish - draft_item", draft_item)
+        # 2- Get its dict form. Since there are no "errors" bc taken directly from
+        # source, we can use draft_item.data. This dict form is what is
+        # serialized out/deserialized in so it should be valid input to load.
+        draft_data = draft_item.data
+        print("RecordDraftService - publish - draft_item.data", draft_item.data)
+        # And here we lost "access" - it became:
+        # {..., 'access': {'embargo': None, 'owned_by': [{}]}}
```

Comment:
but after taking the result_item().data of it, "access" has been changed!
The only thing we did is take a `Schema.dump()` ...

Reply:
So doing

```python
        draft_item = self.result_item(
            self, identity, draft.dumps(), links_config=None
        )
```

to bypass the `access` key/attribute problem and forego the links results in:

```
>>> draft_data
{'expires_at': '2021-02-17T22:54:46.243676+00:00', 'conceptid': 'x84jd-5a641', 'created': '2021-02-17T22:54:46.243640+00:00', 'updated': '2021-02-17T22:54:46.256521+00:00', 'links': {}, 'id': 'kb0qr-yjt60', 'metadata': {}}
```
:thinking: